### PR TITLE
publish directly Plugin portal

### PR DESCRIPTION
**Pull request by Dierk**
Marc, please review and merge if ok.
We should have much less dependencies, quicker publishing, and much simpler use.
Downside: snapshots are never replaced by new versions. We have to bump up the number, then.

### CODE REVIEW

**Comment by mperry** [comment]
Currently fixing a few errors @Dierk.  I have a few questions too.

File: README.adoc [hunk 1]
```diff
@@ -3,34 +3,10 @@
 
 This is the official Gradle plugin to compile Frege projects (https://github.com/Frege/frege).  See the example project, https://github.com/mperry/frege-gradle-example, for how to use this plugin.
 
-The latest release is 0.4.1.  Note that the 0.3 experimented with _github releases_ and http://jitpack.io.  There seems to be access issues to the 0.3 release, please use the 0.4-SNAPSHOT snapshot release.
+= How to refer to on a gradle build
```
Comment (mperry) [inline README.adoc:6]: This language is clumsy and verbose, I prefer "Usage" or "How To Use".

File: README.adoc [hunk 2]
```diff
@@ -3,34 +3,10 @@
...
-
-
-To locally install this plugin use
-
-    gradlew clean install
-
-
-NOTE: There is no need to install gradle or anything else beside a recent Java version.
-      It is all self-installing.
+See https://plugins.gradle.org/plugin/org.frege-lang
```
Comment (mperry) [inline README.adoc]: Fixing this incorrect URL.

File: README.adoc [hunk 3]
```diff
@@ -3,34 +3,10 @@
...
 
 = Continuous Integration
 
-The Travis CI build of this repository is at https://travis-ci.org/Frege/frege-gradle-plugin/                              .
-
-The snapshot builds are automatically deployed to Sonatype at https://oss.sonatype.org/content/groups/public/org/frege-lang/frege-gradle-plugin/.
+The Travis CI build of this repository is at https://travis-ci.org/Frege/frege-gradle-pgitlugin/ .
```
Comment (mperry) [inline README.adoc:12]: Fixing this incorrect URL.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Sun Aug 30 17:39:41 CEST 2015
+#Thu Sep 03 23:57:28 CEST 2015
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.6-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.6-all.zip
```
Comment (mperry) [inline gradle/wrapper/gradle-wrapper.properties:6]: What are the pros and cons of using the bin vs all Gradle distribution?

**Comment by mperry** [comment]
Can we release this as 0.5?  This is the first release with the 3.23 Frege compiler and we are starting to use the Gradle plugin portal.

File: README.adoc [hunk 5]
```diff
@@ -3,34 +3,10 @@
 
 This is the official Gradle plugin to compile Frege projects (https://github.com/Frege/frege).  See the example project, https://github.com/mperry/frege-gradle-example, for how to use this plugin.
```
Comment (mperry) [inline README.adoc:4]: This example project should be in CI and should probably be either an official Frege project or part of this repo.  Your thoughts?

File: build.gradle [hunk 6]
```diff
@@ -30,21 +15,28 @@ ext {
...
     fregeClassifier = "-gaa3af0c"
     fregeVersion = "$fregeBaseVersion$fregeClassifier"
+
+    // work around https://issues.gradle.org/browse/GRADLE-3281
+    def home      = new File(System.getProperty("user.home"))
+    def propfile  = new File(home, ".gradle/gradle.properties")
+    if (propfile.exists()) {
+        props = new Properties()
+        propfile.withReader {
+            props.load(it)
```
Comment (mperry) [inline build.gradle:25]: If the .gradle/gradle.properties file exists, but the key "gradle.publish.key" does not there is a NullPointerException.  This is what I get.

**Comment by mperry** [comment]
Ok, I have the plugin building again now.  Can you review the changes I have made and my questions please @Dierk?

**Comment by mperry** [comment]
Looks like you have published 0.4.3 to https://plugins.gradle.org/plugin/org.frege-lang.  Looks like the last sonatype release was 0.4.1 (https://oss.sonatype.org/content/groups/public/org/frege-lang/frege-gradle-plugin/).  Hmm, can we trash the Gradle plugin portal 0.4.3 and keep 0.4.x to support the other frege projects until 0.5 is released and they start using the plugin portal?

**Comment by Dierk** [comment]
Hi Mark,

> Hmm, can we trash the Gradle plugin portal 0.4.3 and keep 0.4.x to support the other frege projects until 0.5 is released and they start using the plugin portal?

Absolutely.

I just needed a "fresh" version number to give the plugin portal a test run.
I feel it works ok, even though it does not support snapshots (probably for a good reason) and has no other means of "management" e.g. revoking version or updating an exiting one.
The big plus is that the typical user doesn't have to specify any repository.

cheers
Dierk

**Pull request merged**